File: beecrowd/1417 - Liga da Justiça.cs
using System; 
using System.Collections.Generic;

class URI {

	struct Vertex {
		public int v;
		public int degree;
	}

	static int cmp(Vertex a, Vertex b) {
		return b.degree - a.degree;
	}

    static void Main(string[] args) {
		while(true) {

			var l = Console.ReadLine().Split(' ');

			int n = int.Parse(l[0]);
			int m = int.Parse(l[1]);

			if(n + m == 0) break;
	
			Vertex[] S = new Vertex[n];
			int[] preSum = new int[n + 1];

			for(int i = 0; i < n; ++i) {
				S[i].v = i;
				S[i].degree = 0;
			}
			
			for(int i = 0; i < m; ++i) {
				var ll = Console.ReadLine().Split(' ');
				int u = int.Parse(ll[0]) - 1, v = int.Parse(ll[1]) - 1;
				++S[u].degree;
				++S[v].degree;	
			}

			Array.Sort(S, cmp);

			for(int i = 1; i <= n; ++i) preSum[i] = preSum[i - 1] + S[i - 1].degree;

			bool flag = false;

			for(int i = 1; i <= n; ++i)
				flag = flag || preSum[i] == i * (i - 1) + preSum[n] - preSum[i];

			Console.WriteLine(flag ? "Y" : "N");
		}						
    }	
}

File: beecrowd/1417 - Liga da Justiça.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

type Vertex struct {
	vertex int
	degree int
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)

	defer writer.Flush()

	for {
		var n, m int

		fmt.Fscan(reader, &n, &m)

		if n+m == 0 {
			break
		}

		vertexes := make([]Vertex, n)
		prefix := make([]int, n+1)

		for i := 0; i < n; i++ {
			vertexes[i].vertex = i
			vertexes[i].degree = 0
		}

		for i := 0; i < m; i++ {
			var u, v int

			fmt.Fscan(reader, &u, &v)

			vertexes[u-1].degree += 1
			vertexes[v-1].degree += 1
		}

		vertexes_slice := vertexes[:]

		sort.Slice(vertexes_slice, func(i, j int) bool {
			return vertexes_slice[i].degree > vertexes_slice[j].degree
		})

		prefix[0] = 0
		for i := 1; i <= n; i++ {
			prefix[i] = prefix[i-1] + vertexes_slice[i-1].degree
		}

		flag := false

		for i := 1; i <= n; i++ {
			flag = flag || prefix[i] == i*(i-1)+prefix[n]-prefix[i]
		}

		if flag {
			fmt.Fprintln(writer, "Y")
		} else {
			fmt.Fprintln(writer, "N")
		}
	}
}
